tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Example/ExampleTool.cs

```diff
@@ -464,6 +464,26 @@ public async Task<DefaultCommandResponse> DemonstrateMicroagentFibonacci(int n,
                 Instructions = instructions,
                 MaxToolCalls = 7,
                 Tools = [advanceTool],
+                ValidateResult = async result =>
+                {
+                    // Check the result for correctness using fibonacci formula.
+                    var phi = (1 + Math.Sqrt(5)) / 2;
+                    var psi = (1 - Math.Sqrt(5)) / 2;
+                    var expected = (int)Math.Round((Math.Pow(phi, n) - Math.Pow(psi, n)) / Math.Sqrt(5));
+
+                    // Example validation
+                    if (result != expected)
```

Comment:
so this is for deterministic output results (like scripts being run etc.)?

Reply:
Yeah, so taking spelling for example, let's say you are expecting the agent to update the cspell file before it exits. In that case you could add a validate callback which checks the cspell file to make sure that it changed. If it didn't change, then you can provide a reason and it goes back to the agent and it gets to try again.